src/database/index.tsx

```diff
@@ -348,6 +348,46 @@ export const getUser = async (auth0Id: string): Promise<User | null> => {
   }
 };
 
+export const getScenarioUsers = async (
+  scenarioId: string,
+): Promise<Array<User> | null> => {
+  try {
+    const scenario = await getScenario(scenarioId);
+
+    if (!scenario) {
+      throw new Error(`No scenario found matching id ${scenarioId}`);
+    }
+
+    const auth0Ids = Object.keys(scenario.roles);
+
+    const users = [];
+    for (const auth0Id of auth0Ids) {
+      // We have to get the users one at a time because Firestore limits the number
+      // of elements that can be included within an "in" clause to 10. In other
+      // words, if a Scenario has more than 10 users, writing the query with a
+      // where clause that includes each of these user's auth0Ids would fail.
+      // https://firebase.google.com/docs/firestore/query-data/queries#in_and_array-contains-any
+      const user = await getUser(auth0Id);
+
+      if (!user) {
+        throw new Error(
+          "Error retrieving role for ${auth0Id} in scenario {scenarioId}",
+        );
+      }
+
+      users.push(user);
```

Comment:
won't this issue the `getUser` queries serially? I think you could do something like `const users = await Promise.all(auth0Ids.map((auth0Id) => getUser(auth0Id))` if you wanted to fetch them concurrently

Reply:
Oh, super dope.  I didn't even know this was a thing.  I tried it out locally, and after wrangling in TypeScript, everything still functions.  Thanks for helping me level up! 💪